x/leverage/keeper/keeper.go

```diff
@@ -208,26 +222,34 @@ func (k Keeper) BorrowAsset(ctx sdk.Context, borrowerAddr sdk.AccAddress, borrow
 	}
 
 	// Determine amount of all tokens currently borrowed
-	currentlyBorrowed, err := k.GetBorrowerBorrows(ctx, borrowerAddr)
+	borrowed := k.GetBorrowerBorrows(ctx, borrowerAddr)
+
+	// Calculate current borrow limit
+	collateral := k.GetBorrowerCollateral(ctx, borrowerAddr)
+	borrowLimit, err := k.CalculateBorrowLimit(ctx, collateral)
 	if err != nil {
 		return err
 	}
 
-	// Retrieve borrower's account balance.
-	// accountBalance := k.bankKeeper.GetAllBalances(ctx, borrowerAddr)
+	// Calculate borrowed value will be AFTER this borrow
+	newBorrowedValue, err := k.TotalTokenValue(ctx, borrowed.Add(borrow))
+	if err != nil {
+		return err
+	}
 
-	// TODO #213: Use oracle to compute borrow limit and current borrowed value.
-	// Prevent borrows that exceed borrow limit.
+	// Return error if borrowed value would exceed borrow limit
+	if newBorrowedValue.GT(borrowLimit) {
+		return sdkerrors.Wrap(types.ErrBorrowLimitLow, borrowLimit.String())
+	}
 
-	// Note: Prior to oracle implementation, we cannot compare loan value to borrow limit
 	loanTokens := sdk.NewCoins(borrow)
-	if err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, borrowerAddr, loanTokens); err != nil {
+	if err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, borrowerAddr, loanTokens); err != nil {
```

Comment:
You can keep the original code as we probably have no need to shadow the variable.


```suggestion
	if err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, borrowerAddr, loanTokens); err != nil {
```